pkg/frontend/transport/handler.go

```diff
@@ -190,6 +190,17 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	activityIndex := f.at.Insert(func() string { return httpRequestActivity(r, r.Header.Get("User-Agent"), params) })
 	defer f.at.Delete(activityIndex)
 
+	if isStreamingEndpoint(r) {
+		// Disable the write deadline for streaming endpoints.
+		rc := http.NewResponseController(w)
+		err = rc.SetWriteDeadline(time.Time{})
```

Comment:
I'm not familiar with how this works, what happens if a client initiates a request and then just hangs without reading data? Does the connection eventually close or just sit around?

Reply:
Yes, if the deadline is set to the default value the connection will remain open until either
- the response has been transmitted in full
- the client closes it
- some hop in between (e.g. reverse proxy) closes it.

We could instead add a config parameter and use it to set a non-default deadline that's far enough in the future.